<!DOCTYPE html>
    <html>
    <head>
        <meta charset="UTF-8">
        <title>Annotation of TNAU genomes</title>
        <style>
/* From extension vscode.github */
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

.vscode-dark img[src$=\#gh-light-mode-only],
.vscode-light img[src$=\#gh-dark-mode-only] {
	display: none;
}

</style>
        <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex/dist/katex.min.css">
<link href="https://cdn.jsdelivr.net/npm/katex-copytex@latest/dist/katex-copytex.min.css" rel="stylesheet" type="text/css">
        <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/Microsoft/vscode/extensions/markdown-language-features/media/markdown.css">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/Microsoft/vscode/extensions/markdown-language-features/media/highlight.css">
<style>
            body {
                font-family: -apple-system, BlinkMacSystemFont, 'Segoe WPC', 'Segoe UI', system-ui, 'Ubuntu', 'Droid Sans', sans-serif;
                font-size: 14px;
                line-height: 1.6;
            }
        </style>
        <style>
.task-list-item {
    list-style-type: none;
}

.task-list-item-checkbox {
    margin-left: -20px;
    vertical-align: middle;
    pointer-events: none;
}
</style>
        
    </head>
    <body class="vscode-body vscode-light">
        <!----Comments & Links ---->
<!------------------------->
<h1 id="annotation-of-tnau-genomes">Annotation of TNAU genomes</h1>
<p><em>Repeat masking and annotation of the assemblies generated from TNAU S32, S16, S6 and SY-2 isolates.</em></p>
<h2 id="toc">TOC</h2>
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#repeat-masking">Repeat Masking</a>
<ul>
<li><a href="#rename-contigs">Rename Contigs</a></li>
<li><a href="#repeat-modeler">Repeat Modeler</a></li>
<li><a href="#repeatmodeler-results-summary">RepeatModeler Results Summary</a></li>
<li><a href="#repeat-masker">Repeat Masker</a></li>
</ul>
</li>
<li><a href="#annotations">Annotations</a>
<ul>
<li><a href="#maker">MAKER</a>
<ul>
<li><a href="#endnotes">EndNotes</a></li>
</ul>
</li>
</ul>
</li>
</ul>
<hr>
<h1 id="introduction">Introduction</h1>
<p>Collaborators at TNAU aimed to assess the diversity of <em>Fusarium oxysporum</em> f. sp. <em>cubense</em> (<em>Focub</em>) in Tamil Nadu. They conducted a survey in banana-growing areas to identify wilt symptoms in various banana cultivars. Samples with wilt symptoms, including <em>Focub</em> TR4-susceptible 'Grande Nain' bananas, were collected, and the pathogen was isolated and identified as TR4.</p>
<p>Three highly virulent isolates (S6, S16, S32) were identified and sequenced (Illumina), though the DNA extraction protocol used remains unspecified due to communication challenges. Raw reads from these isolates were analysed, and the genomes were assembled using SPAdes. For isolates with high contamination levels, only reads mapping to reference genomes were used for <em>de novo</em> assembly.</p>
<p>This document follows the repeat masking and annotation of these isolates. Annotations will be generated for these assemblies, however, as assemblies are poor quality and no transcriptome data are available, no further analysis will be conducted on the annotations outputs. These will just be used for genome summary statistics.</p>
<hr>
<h1 id="repeat-masking">Repeat Masking</h1>
<p>Followed <a href="(https://darencard.net/blog/2022-07-09-genome-repeat-annotation/)">this tutorial</a> for repeat modeling and masking. I started with the S32 isolate, but the same approach was taken for all isolates: S32, S16, S6 and SY-2. First, repeats were identified in the assemblies <em>de novo</em> using RepeatModeler (version: 2.0.4), followed by repeat masking with RepeatMasker (version: 4.1.5). The RepeatModeler and RepeatMasker install, as well as dependencies can be found here on my Local Machine.</p>
<pre><code class="language-bash">/Users/u1983390/Apps/repeat-annotation
</code></pre>
<p>Note, had install issues with Vettel and it is being decommissioned.</p>
<h2 id="rename-contigs">Rename Contigs</h2>
<p>I renamed the contigs to remove the long names generated by SPAdes, where <code>--nr-width</code> equals the total number of positions in <code>nr</code>. For example, &lt;999 contigs  <code>--nr-width 3</code>; &lt;9999 contigs  <code>--nr-width 4</code> etc.</p>
<pre><code class="language-bash"><span class="hljs-comment">#Check contig number</span>
grep -c <span class="hljs-string">&quot;&gt;&quot;</span> S32_V4-ContaminantFiltered.fasta
2443

<span class="hljs-comment"># Rename contigs</span>
seqkit replace -p <span class="hljs-string">&#x27;.+&#x27;</span> -r <span class="hljs-string">&#x27;Contig_{nr}_TNAU_Fusarium_Isolate_S32&#x27;</span> --nr-width 4  S32_V4-ContaminantFiltered.fasta -o S32_V5-FS66-Contiglabelled.fasta
</code></pre>
<h2 id="repeat-modeler">Repeat Modeler</h2>
<p>All repeat modeling was run in the following directory: <code>[IsolateCode]/RepeatMasking/RepeatModeling</code></p>
<p>First, I built a RepeatModeler database.</p>
<pre><code class="language-bash">BuildDatabase -name S32_Fusarium_SpNov -engine ncbi S32_V5-FS66-Contiglabelled.fasta
</code></pre>
<p>This took only a few seconds and produced the following log:</p>
<pre><code class="language-bash">Building database S32_Fusarium_SpNov: 
Reading S32_V5-FS66-Contiglabelled.fasta... 
Number of sequences (bp) added to database: 2443 ( 40920118 bp )
</code></pre>
<p>This was followed by RepeatModeler, which used the database I had just built.</p>
<pre><code class="language-bash">RepeatModeler -threads 16 -engine ncbi -database S32_Fusarium_SpNov -LTRStruct 2&gt;&amp;1 | <span class="hljs-built_in">tee</span> 00_repeatmodeler.<span class="hljs-built_in">log</span>
</code></pre>
<p>The tutorial does not include the <code>-LTRStruct</code> flag. However, I included it to increase the number of repetative elements found, as these isolates are potentially novel species.</p>
<p>RepeatModeler has produced several output dirs and files. Each dir contains output from each round of RepeatModeler. The tutorial explains that the most relevant outputs are in the output FASTA and Stockholm files:</p>
<blockquote>
<p>The FASTA library contains important repeat classification information in the header lines, which look something like this: &gt;rnd-1_family-174#LINE/L1. A unique round and family identifier are included from the internal RepeatModeler run and after the #, repeat classification information is provided.</p>
</blockquote>
<p>This took approximately 3 hours for S32, and no clean LTR-RTs were kept by LTR_retriever.</p>
<p>[9/11/2023] Again, no clean LTR-RTs were kept by LTR_retriever for S16 - Wondering if the install has not worked for LTR_retriever. However, I think I can continue using this approach and if install has failed, explain it.</p>
<pre><code class="language-bash"><span class="hljs-comment"># S32</span>
grep -c <span class="hljs-string">&quot;&gt;&quot;</span> S32_Fusarium_SpNov-families.fa
39
<span class="hljs-comment"># S16</span>
grep -c <span class="hljs-string">&quot;&gt;&quot;</span> S16_Fusarium_SpNov-families.fa 
57
</code></pre>
<p>Following the tutorial, I added the species code to the head of each element, and split the library into classified and unclassified elements.</p>
<pre><code class="language-bash"><span class="hljs-comment">#Add the prefix:</span>
<span class="hljs-built_in">cat</span> S32_Fusarium_SpNov-families.fa | seqkit fx2tab | awk <span class="hljs-string">&#x27;{ print &quot;FusSpNov-S32_&quot;$0 }&#x27;</span> | seqkit tab2fx &gt; S32_Fusarium_SpNov-families.prefix.fa

<span class="hljs-comment">#Split the library: Classified</span>
<span class="hljs-built_in">cat</span> S32_Fusarium_SpNov-families.prefix.fa | seqkit fx2tab | grep -v <span class="hljs-string">&quot;Unknown&quot;</span> | seqkit tab2fx &gt; S32_Fusarium_SpNov-families.prefix.fa.classified

<span class="hljs-comment">#Split the library: Unclassified</span>
<span class="hljs-built_in">cat</span> S32_Fusarium_SpNov-families.prefix.fa | seqkit fx2tab | grep <span class="hljs-string">&quot;Unknown&quot;</span> | seqkit tab2fx &gt; S32_Fusarium_SpNov-families.prefix.fa.unclassified
</code></pre>
<h4 id="repeatmodeler-results-summary">RepeatModeler Results Summary</h4>
<ul>
<li>For S32, this resulted in 5 classified families and 34 unclassified.</li>
<li>For S16, this resulted in 8 classified families and 49 unclassified.</li>
<li>For S6, this resulted in 50 classified families and 93 unclassified - I am conscious that some may be from the contaminant. This assembly is very fragmented.</li>
<li>For SY-2, this resulted in [X] classified families and [X] unclassified.</li>
</ul>
<p>These unclassified families could be classified further with repeat classifier and an additional TE database, such as Repbase. However, we do not have a RepBase subscription, and curating a further database is time consuming and challenging, considering we are potentially working with a novel species and the assemblies are of poor quality, I do not think it is worth the additional work. If we receive better quality raw data, then we can focus on repeat modeling and masking.</p>
<h2 id="repeat-masker">Repeat Masker</h2>
<p>RepeatMasker will be run repeatedly, with each round using different repeat libraries to annotate the genomes. This is done bt taking the masked output from one round of RepeatMasker, and feeding it into the subsequent round of repeat annotation and masking.</p>
<p>The rounds will progress in the following order:</p>
<ol>
<li>Simple repeats</li>
<li>Curated repeats from a library<sup class="footnote-ref"><a href="#fn1" id="fnref1">[1]</a></sup></li>
<li>Known species-specific repeats from RepeatModeler</li>
<li>Unknown species-specific repeats from RepeatModeler</li>
</ol>
<p>First, we make directories for the output of each run</p>
<pre><code class="language-bash"><span class="hljs-built_in">mkdir</span> -p logs 01_Simple_out 02_Fusarium_out 03_Classified_out 04_Unclassified_out
</code></pre>
<p>Then we begin the first round of simple annotations, where <code>-noint</code> and <code>-xsmall</code> only annotate and soft masks simple repeats.</p>
<pre><code class="language-bash"><span class="hljs-comment"># round 1: </span>
RepeatMasker -pa 16 -a -e ncbi -<span class="hljs-built_in">dir</span> 01_Simple_out -noint -xsmall S32_V5-FS66-Contiglabelled.fasta 2&gt;&amp;1 | <span class="hljs-built_in">tee</span> logs/01_SimpleMask.<span class="hljs-built_in">log</span>
<span class="hljs-comment"># I renamed the output file using &quot;SimpleMask&quot; to keep track, e.g.</span>
 <span class="hljs-built_in">mv</span> 01_Simple_out/S32_V5-FS66-Contiglabelled.fasta.masked 01_Simple_out/S32_V5-FS66-Contiglabelled.SimpleMask.fasta

</code></pre>
<p>Next, I masked Fusarium repeats from the Dfam database. I used the <code>-nolow</code>  to annotate/mask only complex, interspersed repeats (e.g., transposons) and used the <code>-species Fusarium</code> option to identify repeat consensus sequences from Fusarium in the Dfam (version 3.7) database. Theoretically, this would also include RepBase, but we do not have a subscription.</p>
<pre><code class="language-bash"><span class="hljs-comment"># round 2: </span>
RepeatMasker -pa 16 -a -e ncbi -<span class="hljs-built_in">dir</span> 02_Fusarium_out -nolow -species fusarium 01_Simple_out/S32_V5-FS66-Contiglabelled.SimpleMask.fasta 2&gt;&amp;1 | <span class="hljs-built_in">tee</span> logs/02_FusariumMask.<span class="hljs-built_in">log</span>
<span class="hljs-comment">#  I renamed the output file using &quot;FusariumMask&quot; to keep track, e.g.</span>
 <span class="hljs-built_in">mv</span> 02_Fusarium_out/S32_V5-FS66-Contiglabelled.SimpleMask.fasta.masked  02_Fusarium_out/S32_V5-FS66-Contiglabelled.FusariumMask.fasta

</code></pre>
<p>This masked only 50bp of S32, I think probably due to the fragmented nature of the genome and because I was using only the Dfam database.</p>
<p>No repetitive sequences were detected in S16, either, so I had to use 01_SimpleMask fasta as input for S16 repeat masking round 3.</p>
<p>No repetitive sequences were detected in S6, so I had to use 01_Simple_out fasta a input for S6 repeat masking round 3.</p>
<p>The third round of masking makes use of the output from RepeatModeler, first masking the classified elements I identified earlier. Again, I used the <code>-nolow</code>  to annotate/mask only complex, interspersed repeats (e.g., transposons) and the <code>-lib</code> option to define the library ()</p>
<pre><code class="language-bash"><span class="hljs-comment"># round 3:</span>
RepeatMasker -pa 16 -a -e ncbi -<span class="hljs-built_in">dir</span> 03_Classified_out -nolow -lib  RepeatModeling/S32_Fusarium_SpNov-families.prefix.fa.classified 02_Fusarium_out/S32_V5-FS66-Contiglabelled.FusariumMask.fasta 2&gt;&amp;1 | <span class="hljs-built_in">tee</span> logs/03_ClassifiedMask.<span class="hljs-built_in">log</span>

<span class="hljs-comment">#  I renamed the output file using &quot;ClassifiedMask&quot; to keep track, e.g.</span>
 <span class="hljs-built_in">mv</span> 03_Classified_out/S32_V5-FS66-Contiglabelled.FusariumMask.fasta.masked  03_Classified_out/S32_V5-FS66-Contiglabelled.ClassifiedMask.fasta
</code></pre>
<p>The same steps taken in round 3 were taken for round 4, but the unclassified library was used instead, and annotations were added to the output of round 3.</p>
<pre><code class="language-bash"><span class="hljs-comment"># round 4:</span>
RepeatMasker -pa 16 -a -e ncbi -<span class="hljs-built_in">dir</span> 04_Unclassified_out -nolow -lib  RepeatModeling/S32_Fusarium_SpNov-families.prefix.fa.unclassified 03_Classified_out/S32_V5-FS66-Contiglabelled.ClassifiedMask.fasta 2&gt;&amp;1 | <span class="hljs-built_in">tee</span> logs/04_UnclassifiedMask.<span class="hljs-built_in">log</span>

<span class="hljs-comment">#  I renamed the output file using &quot;ClassifiedMask&quot; to keep track, e.g.</span>
 <span class="hljs-built_in">mv</span> 04_Unclassified_out/S32_V5-FS66-Contiglabelled.ClassifiedMask.fasta.masked  04_Unclassified_out/S32_V5-FS66-Contiglabelled.UnclassifiedMask.fasta
</code></pre>
<p>Now all of the outputs are combined to generate the final masked assembly.</p>
<pre><code class="language-bash"><span class="hljs-comment"># Make the directory for the final, combined mask results to sit. </span>
<span class="hljs-built_in">mkdir</span> 05_FullMask_out 

<span class="hljs-comment"># combine full RepeatMasker result files - .cat.gz</span>
<span class="hljs-built_in">cat</span> 01_Simple_out/S32_V5-FS66-Contiglabelled.SimpleMask.fasta.cat.gz 02_Fusarium_out/S32_V5-FS66-Contiglabelled.FusariumMask.fasta.cat.gz 03_Classified_out/S32_V5-FS66-Contiglabelled.ClassifiedMask.fasta.cat.gz 04_Unclassified_out/S32_V5-FS66-Contiglabelled.UnclassifiedMask.fasta.cat.gz &gt; 05_FullMask_out/S32_V5-FS66-Contiglabelled.FullMask.cat.gz

<span class="hljs-comment"># combine RepeatMasker tabular files for all repeats - .out</span>
<span class="hljs-built_in">cat</span> 01_Simple_out/S32_V5-FS66-Contiglabelled.SimpleMask.fasta.out \
&lt;(<span class="hljs-built_in">cat</span> 02_Fusarium_out/S32_V5-FS66-Contiglabelled.FusariumMask.fasta.out | <span class="hljs-built_in">tail</span> -n +4) \
&lt;(<span class="hljs-built_in">cat</span> 03_Classified_out/S32_V5-FS66-Contiglabelled.ClassifiedMask.fasta.out | <span class="hljs-built_in">tail</span> -n +4) \
&lt;(<span class="hljs-built_in">cat</span> 04_Unclassified_out/S32_V5-FS66-Contiglabelled.UnclassifiedMask.fasta.out | <span class="hljs-built_in">tail</span> -n +4) \
&gt; 05_FullMask_out/S32_V5-FS66-Contiglabelled.FullMask.out

<span class="hljs-comment"># copy RepeatMasker tabular files for simple repeats - .out</span>
<span class="hljs-built_in">cat</span> 01_Simple_out/S32_V5-FS66-Contiglabelled.SimpleMask.fasta.out &gt; 05_FullMask_out/S32_V5-FS66-Contiglabelled.SimpleMask.out

<span class="hljs-comment"># combine RepeatMasker tabular files for complex, interspersed repeats - .out</span>
<span class="hljs-built_in">cat</span> 02_Fusarium_out/S32_V5-FS66-Contiglabelled.FusariumMask.fasta.out \
&lt;(<span class="hljs-built_in">cat</span> 03_Classified_out/S32_V5-FS66-Contiglabelled.ClassifiedMask.fasta.out | <span class="hljs-built_in">tail</span> -n +4) \
&lt;(<span class="hljs-built_in">cat</span> 04_Unclassified_out/S32_V5-FS66-Contiglabelled.UnclassifiedMask.fasta.out | <span class="hljs-built_in">tail</span> -n +4) \
&gt; 05_FullMask_out/S32_V5-FS66-Contiglabelled.ComplexMask.out

<span class="hljs-comment"># combine RepeatMasker repeat alignments for all repeats - .align</span>
<span class="hljs-built_in">cat</span> 01_Simple_out/S32_V5-FS66-Contiglabelled.SimpleMask.fasta.align \
02_Fusarium_out/S32_V5-FS66-Contiglabelled.FusariumMask.fasta.align \
03_Classified_out/S32_V5-FS66-Contiglabelled.ClassifiedMask.fasta.align \
04_Unclassified_out/S32_V5-FS66-Contiglabelled.UnclassifiedMask.fasta.align \
&gt; 05_FullMask_out/S32_V5-FS66-Contiglabelled.FullMask.align
</code></pre>
<p>Now I have generated combined files for the <code>.out</code>, <code>.align</code>, and <code>.cat.gz</code> outputs, I will generate a new, combined <code>.tbl</code> file.</p>
<pre><code class="language-bash"><span class="hljs-comment"># summarize repeat compositions from combined analysis of all RepeatMasker rounds</span>
ProcessRepeats -a -species Fusarium 05_FullMask_out/S32_V5-FS66-Contiglabelled.FullMask.cat.gz 2&gt;&amp;1 | <span class="hljs-built_in">tee</span> logs/05_fullmask.<span class="hljs-built_in">log</span>
</code></pre>
<p>Next, we can create GFF3 files for these repeat annotations using a script distributed with RepeatMasker; <code>rmOutToGFF3.pl</code></p>
<pre><code class="language-bash"><span class="hljs-comment"># Note; this was performed in the 05_FullMask_out directory.</span>
 <span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> *.out ; <span class="hljs-keyword">do</span> rmOutToGFF3.pl <span class="hljs-variable">${i}</span> &gt; <span class="hljs-variable">${i}</span>.gff ; <span class="hljs-keyword">done</span>
</code></pre>
<h3 id="endnotes">EndNotes</h3>
<hr class="footnotes-sep">
<section class="footnotes">
<ol class="footnotes-list">
<li id="fn1" class="footnote-item"><p>I used the default Dfam library I had installed with RepeatMasker.
Using Master RepeatMasker Database: /Users/u1983390/Apps/repeat-annotation/RepeatMasker/Libraries/RepeatMaskerLib.h5 | Dfam Version  : 3.7 | Date     : 2023-01-11 | Families : 19,768. <a href="#fnref1" class="footnote-backref">↩︎</a></p>
</li>
</ol>
</section>

        <script async src="https://cdn.jsdelivr.net/npm/katex-copytex@latest/dist/katex-copytex.min.js"></script>
        
    </body>
    </html>