<!DOCTYPE html>
    <html>
    <head>
        <meta charset="UTF-8">
        <title>Prediction of candidate effectors</title>
        <style>
/* From extension vscode.github */
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

.vscode-dark img[src$=\#gh-light-mode-only],
.vscode-light img[src$=\#gh-dark-mode-only] {
	display: none;
}

</style>
        <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex/dist/katex.min.css">
<link href="https://cdn.jsdelivr.net/npm/katex-copytex@latest/dist/katex-copytex.min.css" rel="stylesheet" type="text/css">
        <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/Microsoft/vscode/extensions/markdown-language-features/media/markdown.css">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/Microsoft/vscode/extensions/markdown-language-features/media/highlight.css">
<style>
            body {
                font-family: -apple-system, BlinkMacSystemFont, 'Segoe WPC', 'Segoe UI', system-ui, 'Ubuntu', 'Droid Sans', sans-serif;
                font-size: 14px;
                line-height: 1.6;
            }
        </style>
        <style>
.task-list-item {
    list-style-type: none;
}

.task-list-item-checkbox {
    margin-left: -20px;
    vertical-align: middle;
    pointer-events: none;
}
</style>
        
    </head>
    <body class="vscode-body vscode-light">
        <h1 id="prediction-of-candidate-effectors">Prediction of candidate effectors</h1>
<p>The gene models produced from the MAKER annotations were used to identify candidate effectors in the TNAU datasets. Predicted genes were filtered based on size (&gt;30aa and &lt;450aa), then submitted to SignalP (v4.1), sequences which were predicted to contain a signal peptide were parsed to EffectorP (v2.0.1).</p>
<h2 id="directory-set-up">Directory set up</h2>
<p>I created the <code>Effectors</code> directory in the <code>NewTools-project/exp/</code> directory. A subdirectory was then generated to store symlinks the annotation files, and avoid repeatedly typing lengthy paths.</p>
<pre><code class="language-bash"><span class="hljs-comment"># make the effectors directories </span>
<span class="hljs-built_in">mkdir</span> -p Effectors Effectors/data

<span class="hljs-comment"># create symlink for annotations. </span>

<span class="hljs-built_in">ln</span> -s ../../GenomeAnnotations/S6/S6_V4.2-FocR1.Contiglabelled.FullMask.maker.output/S6_V4.2-FocR1.Contiglabelled.FullMask.all.maker.proteins.fasta ./
<span class="hljs-built_in">ln</span> -s ../../GenomeAnnotations/S16/S16_V4-Contiglabelled.FullMask.maker.output/S16_V4-Contiglabelled.FullMask.all.maker.proteins.fasta ./
<span class="hljs-built_in">ln</span> -s ../../GenomeAnnotations/S32/S32_V5-FS66-Contiglabelled.FullMask.maker.output/S32_V5-FS66-Contiglabelled.FullMask.all.maker.proteins.fasta ./
<span class="hljs-built_in">ln</span> -s ../../GenomeAnnotations/SY-2/SY-2_V5-RepeatMasked.maker.output/SY-2_V5-RepeatMasked.all.maker.proteins.fasta ./
</code></pre>
<h2 id="size-filtering">Size filtering</h2>
<p>Once the annotation files had been linked, i reduced the total number of candidates from each by filtering the effectors based on size.</p>
<pre><code class="language-bash"><span class="hljs-comment"># generate the processing directory.</span>
<span class="hljs-built_in">mkdir</span> sizeFilter

<span class="hljs-comment"># generate index file for size filtering.</span>
<span class="hljs-built_in">cd</span> data 
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> * ; <span class="hljs-keyword">do</span> <span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span> ; samtools faidx <span class="hljs-variable">${i}</span> ; <span class="hljs-keyword">done</span> 

<span class="hljs-comment"># filter all sequences by size and create sorted bed file of short sequences . &lt;450aa and &gt;30aa accepted. </span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> *.fai ; <span class="hljs-keyword">do</span> <span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;<span class="hljs-variable">${i}</span>...&quot;</span> ; awk <span class="hljs-string">&#x27;{if($2 &lt; 450 &amp;&amp; $2 &gt; 30) print $1 &quot;\t0\t&quot; $2 &quot;\t&quot;}&#x27;</span> <span class="hljs-variable">${i}</span> &gt; ../sizeFilter/<span class="hljs-variable">${i}</span>.sizeFilter.bed ; bedtools <span class="hljs-built_in">sort</span> -i ../sizeFilter/<span class="hljs-variable">${i}</span>.sizeFilter.bed &gt; ../sizeFilter/<span class="hljs-variable">${i}</span>.sizeFilter.sorted.bed ; <span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;done.&quot;</span> ; <span class="hljs-keyword">done</span>

<span class="hljs-comment"># generate fasta of filtered sequences</span>
<span class="hljs-comment">#---</span>
<span class="hljs-comment"># S6</span>
bedtools getfasta -s -<span class="hljs-keyword">fi</span> S6_V4.2-FocR1.Contiglabelled.FullMask.all.maker.proteins.fasta -bed ../sizeFilter/S6_V4.2-FocR1.Contiglabelled.FullMask.all.maker.proteins.fasta.fai.sizeFilter.sorted.bed -fo ../sizeFilter/S6_V4.2-FocR1.Contiglabelled.FullMask.all.maker.proteins.sizeFilter.sorted.fasta

<span class="hljs-comment"># S16 </span>
bedtools getfasta -s -<span class="hljs-keyword">fi</span> S16_V4-Contiglabelled.FullMask.all.maker.proteins.fasta -bed ../sizeFilter/S16_V4-Contiglabelled.FullMask.all.maker.proteins.fasta.fai.sizeFilter.sorted.bed -fo ../sizeFilter/S16_V4-Contiglabelled.FullMask.all.maker.proteins.sizeFilter.fasta

<span class="hljs-comment"># S32</span>
bedtools getfasta -s -<span class="hljs-keyword">fi</span> S32_V5-FS66-Contiglabelled.FullMask.all.maker.proteins.fasta -bed ../sizeFilter/S32_V5-FS66-Contiglabelled.FullMask.all.maker.proteins.fasta.fai.sizeFilter.sorted.bed -fo ../sizeFilter/S32_V5-FS66-Contiglabelled.FullMask.all.maker.proteins.sizeFilter.fasta

<span class="hljs-comment"># SY-2</span>
bedtools getfasta -s -<span class="hljs-keyword">fi</span>  SY-2_V5-RepeatMasked.all.maker.proteins.fasta -bed ../sizeFilter/SY-2_V5-RepeatMasked.all.maker.proteins.fasta.fai.sizeFilter.sorted.bed -fo ../sizeFilter/SY-2_V5-RepeatMasked.all.maker.proteins.sizeFilter.fasta

<span class="hljs-built_in">cd</span> ../sizeFilter/

<span class="hljs-comment"># check the number of short sequences for each isolate</span>
grep -c <span class="hljs-string">&quot;&gt;&quot;</span> *.fasta
S16_V4-Contiglabelled.FullMask.all.maker.proteins.sizeFilter.fasta:9148
S32_V5-FS66-Contiglabelled.FullMask.all.maker.proteins.sizeFilter.fasta:9253
S6_V4.2-FocR1.Contiglabelled.FullMask.all.maker.proteins.sizeFilter.sorted.fasta:10946
SY-2_V5-RepeatMasked.all.maker.proteins.sizeFilter.fasta:9115
</code></pre>
<h2 id="signalp">Signalp</h2>
<p>I created a directory for the analysis, <code>mkdir signalp</code>. The size filtered fasta files were then symlinked in the <code>signalp</code> directory.</p>
<p>I  filtered the short sequences through signalp (v5) using the following shell script <code>signalp.sh</code>.</p>
<pre><code class="language-bash"><span class="hljs-meta">#!/bin/bash</span>

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> *.fasta ; <span class="hljs-keyword">do</span> 
    <span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;Processing <span class="hljs-variable">${i}</span>...&quot;</span> ;  
    signalp -fasta ./<span class="hljs-variable">${i}</span> 1&gt;./<span class="hljs-variable">${i}</span>-signalp.out 2&gt;./<span class="hljs-variable">${i}</span>-signalp.log <span class="hljs-comment">#Identify sequences with signal peptide.</span>
    <span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;Generating index files...&quot;</span> ;
    awk <span class="hljs-string">&#x27;$2 ~ /SP\(Sec\/SPI\)/ {print $1}&#x27;</span> ./<span class="hljs-variable">${i}</span>/<span class="hljs-variable">${i}</span>-Translated_summary.signalp5 &gt; ./<span class="hljs-variable">${i}</span>-signalPepPositives.list;
    <span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;Generating fasta...&quot;</span>;
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(<span class="hljs-built_in">cat</span> ./<span class="hljs-variable">${i}</span>-signalPepPositives.list); <span class="hljs-keyword">do</span> 
        samtools faidx ./<span class="hljs-variable">${i}</span> <span class="hljs-variable">${j}</span>; 
        <span class="hljs-keyword">done</span> &gt; ./<span class="hljs-variable">${i}</span>-signalp.filtered.fasta;
    <span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;done.&quot;</span> ;
<span class="hljs-keyword">done</span>
</code></pre>
<p>The shell script was run from the command line as follows,</p>
<pre><code class="language-bash"><span class="hljs-comment"># run using this command</span>
../../bin/signalp.sh 2&gt;1&amp; | <span class="hljs-built_in">tee</span> signalp.log
</code></pre>
<h2 id="effectorp">EffectorP</h2>
<p>Likely effectors were then predicted from this small secreted protein set using EffectorP (v2.0.1), run in the <code>signalp</code> directory using the following command:</p>
<pre><code class="language-bash"><span class="hljs-comment"># make directory for effectorp output</span>
<span class="hljs-built_in">mkdir</span> ../effectorp

<span class="hljs-comment"># run effectorp</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> *-signalp.filtered.fasta ; <span class="hljs-keyword">do</span> \
    <span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;processing <span class="hljs-variable">${i}</span>...&quot;</span> ; \
    EffectorP.py -i ./<span class="hljs-variable">${i}</span> -E ../effectorp/<span class="hljs-variable">${i}</span>-EffectorP.filtered.fasta &gt; ../effectorp/<span class="hljs-variable">${i}</span>-EffectorP.filtered.log ; \
    <span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;done.&quot;</span> ; \
<span class="hljs-keyword">done</span> 
</code></pre>
<p>I checked the total number of candidates per assembly using grep.</p>
<pre><code class="language-bash"><span class="hljs-built_in">cd</span> ../effectorp/
grep -c  <span class="hljs-string">&quot;&gt;&quot;</span> *.fasta
S16_V4-Contiglabelled.FullMask.all.maker.proteins.sizeFilter.fasta-signalp.filtered.fasta-EffectorP.filtered.fasta:289
S32_V5-FS66-Contiglabelled.FullMask.all.maker.proteins.sizeFilter.fasta-signalp.filtered.fasta-EffectorP.filtered.fasta:314
S6_V4.2-FocR1.Contiglabelled.FullMask.all.maker.proteins.sizeFilter.sorted.fasta-signalp.filtered.fasta-EffectorP.filtered.fasta:333
SY-2_V5-RepeatMasked.all.maker.proteins.sizeFilter.fasta-signalp.filtered.fasta-EffectorP.filtered.fasta:289
</code></pre>
<p>As the file names have become very large, I renamed the fastas.</p>
<pre><code class="language-bash"><span class="hljs-comment"># S6</span>
<span class="hljs-built_in">mv</span> S6_V4.2-FocR1.Contiglabelled.FullMask.all.maker.proteins.sizeFilter.sorted.fasta-signalp.filtered.fasta-EffectorP.filtered.fasta S6_V4.2-candidateEffectors.fasta

<span class="hljs-comment"># S16</span>
<span class="hljs-built_in">mv</span> S16_V4-Contiglabelled.FullMask.all.maker.proteins.sizeFilter.fasta-signalp.filtered.fasta-EffectorP.filtered.fasta S16_V4-candidateEffectors.fasta

<span class="hljs-comment"># S32</span>
<span class="hljs-built_in">mv</span> S32_V5-FS66-Contiglabelled.FullMask.all.maker.proteins.sizeFilter.fasta-signalp.filtered.fasta-EffectorP.filtered.fasta S32_V5-FS66-candidateEffectors.fasta

<span class="hljs-comment"># SY-2</span>
<span class="hljs-built_in">mv</span> SY-2_V5-RepeatMasked.all.maker.proteins.sizeFilter.fasta-signalp.filtered.fasta-EffectorP.filtered.fasta SY-2_V5-candidateEffectors.fasta
</code></pre>
<p>I then clustered the candidate effectors at 80% identity to identify shared candidates. Initially, i had to create a single fasta.</p>
<pre><code class="language-bash"><span class="hljs-comment"># create an empty file to hold all of the candidate effectors.</span>
<span class="hljs-built_in">touch</span> ./AllCandidateEffectorSets.fasta 
<span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;Clustering final effector sets...&quot;</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> *.fasta; 
<span class="hljs-comment"># Add genome filename to start of Fasta Headers so we know which isolate this came from and  combine all of the individual candidate effector sets by adding them to the empty candidate effector fasta.</span>
  <span class="hljs-keyword">do</span>
  awk <span class="hljs-string">&#x27;/&gt;/{sub(&quot;&gt;&quot;,&quot;&amp;&quot;FILENAME&quot;_&quot;);sub(/\.fasta/,x)}1&#x27;</span> <span class="hljs-variable">${i}</span> &gt;&gt; AllCandidateEffectorSets.fasta ; 
<span class="hljs-keyword">done</span> 
</code></pre>
<p>Then cluster the effectors by sequence identity (80%).</p>
<pre><code class="language-bash"><span class="hljs-comment"># make a directory for the cdhit output</span>
<span class="hljs-built_in">mkdir</span> cdhit

<span class="hljs-comment"># symlink the cdhit input file (from the effectorp dir).</span>
<span class="hljs-built_in">ln</span> -s AllCandidateEffectorSets.fasta ../cdhit/

cdhit -i ../effectorp/AllCandidateEffectorSets.fasta -d 0 -o ./AllCandidateEffectorSets -c 0.80 -n 5  -G 1 -g 0 -b 20 -l 10 -s 0.0 -aL 0.0 -aS 0.0 1&gt; cd-hit.log
</code></pre>
<p>I processed the <code>cd-hit</code> data using a custom python script, <a href="https://github.com/JamiePike/NewTools-Project/blob/master/bin/ProcessingCdhit.py">ProcessingCdhit.py</a>.</p>
<hr>
<p>Going forward...</p>
<ul>
<li>BLAST against focub genomes?</li>
<li>BLAST focub set against these assemblies?</li>
</ul>

        <script async src="https://cdn.jsdelivr.net/npm/katex-copytex@latest/dist/katex-copytex.min.js"></script>
        
    </body>
    </html>